src/app/analyze/layout.tsx

```diff
@@ -53,19 +58,74 @@ const Layout = ({ children }: LayoutProps) => {
     }
   );
 
-  const tooltipIndex = (STEPS.find((v) => v.route === pathname)?.id ?? 1) - 1;
-
   const TOOLTIP_CONTENTS = [
     <>“OOO님 좋은 시작이에요”</>,
     <>“뭐든지 시작이 반이에요”</>,
     <>“내용을 풍부하게 작성할수록 다양한 AI 직무 역량 키워드를 받을 수 있어요 최선을 다해 작성해주세요 :)”</>,
     <>“AI 직무역량 키워드 추천을 통해 나의 직무역량을 다각도로 넓혀보세요”</>,
   ];
 
+  const setWriteStatus = useCallback(
+    (target: WriteStatusType[], status: WriteStatusType) => {
+      target[prevStepIndex] = status;
+      methods.setValue('writeStatus', target);
+    },
+    [methods, prevStepIndex]
+  );
+
   useEffect(() => {
-    // TODO: switch문으로 현재 페이지에 해당하는 값들이 모두 있으면 작성 완료, 모두 만족하진 않고 있긴하면 작성중, 없으면 미작성,
-    // ['미작성', '작성중', '미작성', '작성완료']
-  }, [pathname]);
+    const writeStatus = methods.getValues('writeStatus') as WriteStatusType[];
+    const copyWriteStatus = [...writeStatus];
+
+    switch (prevPathname) {
+      case ROUTES.EXPERIENCE:
+        const [title, startYYYY, startMM, endYYYY, endMM, experienceRole, motivation] = methods.getValues([
+          'title',
+          'startYYYY',
+          'startMM',
+          'endYYYY',
+          'endMM',
+          'experienceRole',
+          'motivation',
+        ]);
+        if (!!title && !!startYYYY && !!startMM && !!endYYYY && !!endMM && !!experienceRole && !!motivation) {
+          setWriteStatus(copyWriteStatus, '작성완료');
+        } else if (!!title || !!startYYYY || !!startMM || !!endYYYY || !!endMM || !!experienceRole || !!motivation) {
+          setWriteStatus(copyWriteStatus, '작성중');
+        } else {
+          setWriteStatus(copyWriteStatus, '미작성');
+        }
+        break;
+      case ROUTES.KEYWORD:
+        const keyword = methods.getValues('keyword');
+        if (keyword.some(([, isSelected]) => isSelected === true)) {
+          setWriteStatus(copyWriteStatus, '작성완료');
+        } else {
+          setWriteStatus(copyWriteStatus, '미작성');
+        }
+        break;
+      case ROUTES.INFORMATION:
+        const [situation, task, action, result] = methods.getValues(['situation', 'task', 'action', 'result']);
+        if (!!situation && !!task && !!action && !!result) {
+          setWriteStatus(copyWriteStatus, '작성완료');
+        } else if (!!situation || !!task || !!action || !!result) {
+          setWriteStatus(copyWriteStatus, '작성중');
+        } else {
+          setWriteStatus(copyWriteStatus, '미작성');
+        }
```

Comment:
이 로직이 계속 중복되니까 함수로 분리해도 좋을 것 같습니다~

Reply:
음 저도 그 생각을 해보았는데요, 가독성이 오히려 더 안좋아질까봐 분리하지 않았습니다! 혹시 가독성을 해치지 않고 분리하면 오히려 더 가독성이 좋아질 수 있는 방법이 있다면 공유부탁드립니다!

Reply:
```suggestion
        const formValues = methods.getValues(['situation', 'task', 'action', 'result']);
        const writeStatus = getWriteStatus(formValues);

        setWriteStatus(copyWriteStatus, writeStatus)
```

이렇게 해볼 수 있지 않을까요?! 

반환 값으로 set 해주기만 하면 되는 방식으로요!

Reply:
getWriteStatus() 내부 구현에서 일관성 있는 분기가 되면 상관없는데 string, boolean이 아닌 객체 형식은 length를 비교하거나 isEmpty()함수를 사용해서 비었는지를 판별해야해서 함수로 사용하기 까다로울것이라고 판단했습니다! 혹시 공감이 되실까요~? 자유롭게 의견 부탁드립니다!

Reply:
아~ 그렇네요! getWriteStatus()이 복잡해지겠네유.. 굳이 분리하지 않는게 좋을 것 같습니다! 👍